src/DelegatingContainer.php

```diff
@@ -38,43 +38,88 @@ public function __construct(ServiceProviderInterface $provider, PsrContainerInte
      * {@inheritDoc}
      */
     public function get($id)
+    {
+        static $stack = [];
```

Comment:
What is the advantage of using a `static` variable VS property?

Reply:
Functionally equivalent.

The advantages for me are of a practical and functional nature, but there are conceptual reasons as well, since I know you love those ;)

I choose static variables to keep all relevant data in one place, making it more of a closed-box, which in turn makes it more portable, which in turn makes it more reusable, should you want to make the function a standalone thing that implementations use.

You can also think of it as a property belonging to a class while a static variable belongs to a function. Belonging to a class means it has relevance in the overarching concept of a `DelegatingContainer` or a key-resolving-factory-calling-extension-supporting container in general, which I think we can agree it doesn't.

Reply:
Seems like this implementation relies on every actual access to a service being done via an instance of the same class: every time _its_ `get()` method runs, the stack will have a service name pushed onto it. So from my perspective, it is a stack of retrievals from the `DelegatingContainer`, and therefore pertains to the class instance.

Reply:
The bigger problem is that `static` inside a class method will point to the same value _for any instance of that class_. So if somewhere in a service definition a `get()` method of an entirely different `DelegatingContainer` is used, this will push onto the stack.

Reply:
✅ 92c5bfd